src-self-hosted/link.zig

```diff
@@ -2812,6 +2828,100 @@ pub const File = struct {
         }
 
     };
+
+    pub const MachO = struct {
+        pub const base_tag: Tag = .macho;
+
+        base: File,
+
+        ptr_width: enum { p32, p64 },
```

Comment:
I think... can we assume that MachO is macos-specific and therefore always 64 bit? https://support.apple.com/en-us/HT208436